.github/workflows/release-sources.yml

```diff
@@ -0,0 +1,59 @@
+name: Release Sources
+
+permissions:
+  contents: read
+
+on:
```

Comment:
Are we able to add a `pull_request` trigger that does everything except for uploading the final artifact to verify that this works when making modifications?

I think that would require moving the check permissions step, but I don't think running `export.sh` or the provenance attestation need those permissions?

Reply:
I added a pull_request trigger.  Let me know what you think I had to disable the attestation step, because it doesn't have the right permissions when running from a fork.

Reply:
@boomanaiden154 What do you think about this pull_request trigger?  Is it worth having even if it's limited in what it can test.

Reply:
I'm going to try to merge this tomorrow, so I can use it out for the upcoming 18.1.8 release.  The pull_request trigger does seem useful even if it is somewhat invasive.

Reply:
It will probably catch something at some point, but definitely doesn't test the full functionality. Current changes seem fine to me.